### PR TITLE
Refactor engine-gecko tests to remove references to GeckoBundle

### PR DESCRIPTION
**Pull request by csadilek**
Removes references to `GeckoBundle` and makes sure we don't subclass any `GeckoView` classes. Finished this for notifications, selection, and extensions. With this we can remove our overridden `org.mozilla.geckoview` package now.

~~There's one piece remaining: `GeckoPromptDelegateTest` which we can do in a follow-up.~~ Updated to refactor those tests as well.

### CODE REVIEW

File: components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/webnotifications/MockWebNotification.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    title: String,
+    tag: String,
+    text: String,
+    imageUrl: String,
+    textDirection: String,
+    lang: String,
+    requireInteraction: Boolean,
+    source: String? = null,
+    silent: Boolean = false,
+    vibrate: IntArray = IntArray(0)
```
Comment (agi) [inline components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/webnotifications/MockWebNotification.kt:21]: FWIW most of these are nullable, not sure if we care: https://searchfox.org/mozilla-central/source/mobile/android/geckoview/api.txt#2050-2059

**Review by agi — APPROVED** [review]
Looks good to me! Thank you

File: components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/webnotifications/MockWebNotification.kt (hunk 1 above)
Comment (csadilek) [reply components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/webnotifications/MockWebNotification.kt]: OK, updated to align with GV API.

**Comment by csadilek** [comment]
Updated to include all tests.

**Comment by agi** [comment]
Wow thanks! I looked over the prompt changes and those look good too.

**Review by pocmo — APPROVED** [review]

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**